### PR TITLE
Add class level validation

### PR DESCRIPTION
**Pull request by rlvandaveer**
This is a PR related to issue #211. This is a work in progress and I am submitted it to solicit feedback before adding any unit tests.

----

Add class level validation attribute to attribute validation sample
Modify CommandAttributeConvention to discover class level validation attributes
Modify IValidator to inherit from ICommandValidator
Implement ICommandValidator in AttributeValidator

### CODE REVIEW

File: docs/samples/validation/attributes/Program.cs [hunk 1]
```diff
@@ -3,8 +3,10 @@
 
 using System;
 using System.ComponentModel.DataAnnotations;
+using System.Linq;
 using McMaster.Extensions.CommandLineUtils;
 
+[RequiredDependentsValidation(nameof(Attachments), nameof(MaxSize))]
```
Comment (natemcmaster) [inline docs/samples/validation/attributes/Program.cs]: This sample was a little hard to follow.  I think what you're trying to do is assert that if `--max-size` is specified, `--attachments` must also have been given, right?

If so, rather than writing an abstract validation attribute, let's make this validation attribute very specific to what we want to validate. It makes it easier to understand. I'm thinking something like `[MaxSizeOptionRequiresAttachments]` with an implementation like

```
protected override ValidationResult IsValid(object value, ValidationContext context)
{
   if (value is AttributeProgram obj)
   {
      	if (obj.MaxSize.HasValue && Attachments == null || Attachments.Length == 0)
     	{
        	return new ValidationResult("--max-size cannot be used unless --attachments is also specified");
    	}
   	}
	return ValidationResult.Success;
}
```

Comment (rlvandaveer) [reply docs/samples/validation/attributes/Program.cs]: Regarding the sample, I can see you point. If it was for real life I would make something more reusable. I'll make this change.

**Comment by rlvandaveer** [comment]
Added a unit test to confirm that class level validation will fire for an application model.
